node/popn.go

```diff
@@ -1239,6 +1285,36 @@ func (nd *node) Add(ctx context.Context, dag ipldformat.DAGService, buf io.Reade
 	return n.Cid(), nil
 }
 
+// selectDAGParams returns the best chunk params according to the file's type
+func selectDAGParams(filepath string, buf io.ReadSeeker) (
```

Comment:
If you do something like:
```suggestion
func selectDAGOptions(filepath string, buf io.ReadSeeker) []DAGOption
```